apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/EndpointJsonComparatorBase.java

```diff
@@ -0,0 +1,163 @@
+package gov.cms.bfd.server.war;
+
+import static io.restassured.RestAssured.given;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import gov.cms.bfd.model.rif.entities.Beneficiary;
+import gov.cms.bfd.model.rif.samples.StaticRifResourceGroup;
+import gov.cms.bfd.server.war.utils.AssertUtils;
+import io.restassured.http.Header;
+import io.restassured.http.Headers;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/** Holds shared functionality between endpoint comparator tests in v1/v2. */
+public abstract class EndpointJsonComparatorBase extends ServerRequiredTest {
+
+  /** Test to use for an ignored field. */
+  protected static final String IGNORED_FIELD_TEXT = "IGNORED_FIELD";
+
+  /**
+   * Generates current endpoint response files, comparing them to the corresponding approved
+   * responses.
+   *
+   * <p>If you've recently made mapping changes, and this test fails, you may need to re-generate
+   * the expected response files by building the project with the maven parameter
+   * -DgenerateTestData=true which will run the test that generates new response files. This should
+   * only be done if you're sure your mapping changes are correct, as these files are used for
+   * regression testing future changes. Once generated, these newly minted files should be checked
+   * in with your mapping changes.
```

Comment:
Added a blurb here to help people who make mapping changes and break this test without knowing they need to regenerate the regression files! There's also some info at the top of the test classes.